### PR TITLE
handle missing 'terminated' key for tasksteps in get_error_message

### PR DESCRIPTION
**Pull request by rcerven**
* CLOUDBLD-9865

Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a JSON/YAML configuration changes are updated in the relevant schema
- n/a Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

File: osbs/tekton.py [hunk 1]
```diff
@@ -498,14 +498,19 @@ def get_error_message(self):
...
+                        exit_code = step['terminated']['exitCode']
+                        if exit_code == 0:
+                            continue
+
+                        reason = step['terminated']['reason']
+                        err_message += f"task step '{step['name']}' failed with exit " \
+                                       f"code: {exit_code} " \
+                                       f"and reason: '{reason}'"
+                    else:
+                        err_message += f"task step '{step['name']}' is missing 'terminated' key " \
```
Comment (chmeliik) [inline osbs/tekton.py:511]: If we logged something here about the taskRun status, could it help us figure  out why this is happening? Or is this one of those things that are logged only in koji-containerbuild that we'll never see?

**Review by chmeliik — COMMENTED** [review]
LGTM, just wondering if we could do something to help debug the problem eventually

File: osbs/tekton.py (hunk 1 above)
Comment (rcerven) [reply osbs/tekton.py:511]: task run status doesn't contain anything useful apart from succeeded or failed, if you want details you need to check steps, which is this code doing

**Review by mkosiarc — APPROVED** [review]

**Comment by rcerven** [comment]
no rn update, fix for unreleased change

**Pull request merged**